driver-async/src/test/functional/com/mongodb/async/client/InitialDnsSeedlistDiscoveryTest.java

```diff
@@ -96,48 +94,35 @@ public void shouldResolve() {
-                    assertEquals(entry.getValue().asNumber().intValue(), (int) connectionString.getConnectTimeout());
-                } else if (entry.getKey().equals("replicaSet")) {
+                if (entry.getKey().equals("replicaSet")) {
                     assertEquals(entry.getValue().asString().getValue(), connectionString.getRequiredReplicaSetName());
-                } else if (entry.getKey().equals("socketTimeoutMS")) {
-                    assertEquals(entry.getValue().asNumber().intValue(), (int) connectionString.getSocketTimeout());
-                } else if (entry.getKey().equals("readPreference")) {
-                    assertEquals(entry.getValue().asString().getValue(), connectionString.getReadPreference().getName());
-                } else if (entry.getKey().equals("readPreferenceTags")) {
-                    assertEquals(asTagSetList(entry.getValue().asArray()),
-                            ((TaggableReadPreference) connectionString.getReadPreference()).getTagSetList());
+                } else if (entry.getKey().equals("ssl")) {
+                    assertEquals(entry.getValue().asBoolean().getValue(), connectionString.getSslEnabled());
+                } else if (entry.getKey().equals("authSource")) {
+                    // ignoring authSource for now, because without at least a userName also in the connection string,
+                    // the authSource is ignored.  If the test gets this far, at least we know that a TXT record
+                    // containing in authSource doesn't blow up.  We just don't test that it's actually used.
+                    assertTrue(true);
                 } else {
                     throw new UnsupportedOperationException("No support configured yet for " + entry.getKey());
                 }
             }
         }
     }
 
-    private List<TagSet> asTagSetList(final BsonArray bsonArray) {
-        List<TagSet> retVal = new ArrayList<TagSet>(bsonArray.size());
-        for (BsonValue cur : bsonArray) {
-            BsonDocument curDocument = cur.asDocument();
-            List<Tag> tagList = new ArrayList<Tag>(curDocument.size());
-            for (Map.Entry<String, BsonValue> curEntry : curDocument.entrySet()) {
-                tagList.add(new Tag(curEntry.getKey(), curEntry.getValue().asString().getValue()));
-            }
-            retVal.add(new TagSet(tagList));
-        }
-        return retVal;
-    }
-
     @Test
     public void shouldDiscover() throws InterruptedException {
         if (seeds.isEmpty()) {
             return;
         }
-        assumeTrue(isDiscoverableReplicaSet());
-
         final CountDownLatch latch = new CountDownLatch(1);
 
         ConnectionString connectionString = new ConnectionString(uri);
 
+        SslSettings sslSettings = getSslSettings(connectionString);
+
+        assumeTrue(isDiscoverableReplicaSet() && getSslSettings().isEnabled() == sslSettings.isEnabled());
```

Comment:
The end result is that the test URIs which resolve to ssl=true will be tested here when running against an SSL-enable replica set, and the ones which resolve to ssl=false will be tested here when running against an SSL-disable replica set.